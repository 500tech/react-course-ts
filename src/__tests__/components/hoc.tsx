import * as React from 'react';
import withTestId from '../../components/hoc';
import {mount} from "enzyme";

describe('HOC: withTestId', () => {
  test('should render component with test id wrapper', () => {
    const TestComponent = () => <div>Test</div>;
    const WrappedComponent = withTestId(TestComponent, 'my-id');

    const component = mount(
        <WrappedComponent />
    );

    expect(component).toMatchSnapshot();
  });
});
