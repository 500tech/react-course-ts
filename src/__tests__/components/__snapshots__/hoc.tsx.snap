// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOC: withTestId should render component with test id wrapper 1`] = `
<Component>
  <div
    data-test-id="my-id"
  >
    <TestComponent>
      <div>
        Test
      </div>
    </TestComponent>
  </div>
</Component>
`;
